uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Optional, Tuple, Type  # noqa: F401
+
+from uaclient import actions, contract, entitlements, event_logger, messages
+from uaclient.api import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.clouds import AutoAttachCloudInstance  # noqa: F401
+from uaclient.config import UAConfig
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.entitlements.base import UAEntitlement  # noqa: F401
+from uaclient.entitlements.entitlement_status import CanEnableFailure
+
+event = event_logger.get_event_logger()
+
+
+class FullAutoAttachOptions(DataObject):
+    fields = [
+        Field("enable", data_list(StringDataValue), False),
+        Field("enable_beta", data_list(StringDataValue), False),
+    ]
+
+    def __init__(
+        self,
+        enable: Optional[List[str]] = None,
+        enable_beta: Optional[List[str]] = None,
+    ):
+        self.enable = enable
+        self.enable_beta = enable_beta
+
+
+class FullAutoAttachResult(DataObject, AdditionalInfo):
+    pass
+
+
+def _get_auto_attach_token(cfg: UAConfig, cloud: AutoAttachCloudInstance):
+    contract_client = contract.UAContractClient(cfg)
+    try:
+        tokenResponse = contract_client.request_auto_attach_contract_token(
+            instance=cloud
+        )
+    except exceptions.ContractAPIError as e:
+        if e.code and 400 <= e.code < 500:
+            raise exceptions.NonAutoAttachImageError(
+                messages.UNSUPPORTED_AUTO_ATTACH
+            )
+        raise e
+
+    token = tokenResponse["contractToken"]
+    return token
+
+
+def _is_any_beta(cfg: UAConfig, ents: List[str]) -> Tuple[bool, str]:
+    for name in ents:
+        try:
+            ent_cls = entitlements.entitlement_factory(cfg, name)
+            if ent_cls.is_beta:
+                return True, name
+        except exceptions.EntitlementNotFoundError:
+            continue
+    return False, ""
+
+
+def _is_compatible_services_present(
+    cfg, ent_list: List[str]
+) -> Tuple[bool, str, List[str]]:
+    ent_cls = list()  # type: List[Type[UAEntitlement]]
+    for e in ent_list:
+        ent_cls.append(entitlements.entitlement_factory(cfg, e))
+
+    for ent in ent_cls:
+        if ent().incompatible_services:
```

Comment:
Let's instantiate the entitlement only once in the code. Additionally, let's pass the `cfg` object to it, otherwise we will unnecessarily create another cfg object when creating the object.